### ISSUE TITLE
Improve vertical-align detection in declaration-block-no-ignored-properties

### ISSUE THREAD

**Issue by efedorenko** [stylelint/stylelint#1204]
At the moment `vertical-align` is disallowed only when used with property `display: block`, but according to spec and browsers implementations it is ignored in many more cases. This property can be applied [only to inline-level and `table-cell` elements](http://www.w3.org/TR/CSS2/visudet.html#propdef-vertical-align), which excludes such `display` values as `list-item`, `table`, `table-row-group`, `table-column`, `table-column-group`, `table-header-group`, `table-footer-group`, `table-row`, and `table-caption`. Applying only to inline-level elements [also means](http://www.w3.org/TR/CSS2/visuren.html#dis-pos-flo) it can't be used with `position: absolute` or `fixed`, and `float: left` or `right`.

`declaration-block-no-ignored-properties` should warn when it finds:

```css
a {
  float: right;
  vertical-align: baseline;
}
```

```css
a {
  position: absolute;
  vertical-align: baseline;
}
```

```css
table {
  display: table;
  vertical-align: baseline;
}
```

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/1205
